src-plugins/manualRegistration/manualRegistrationToolBox.cpp

```diff
@@ -332,45 +319,63 @@ void manualRegistrationToolBox::synchroniseMovingFuseView()
 {
-    if (!d->controller)
-        return;
+    if (d->controller)
+    {
+        if (d->controller->checkLandmarks() != DTK_FAILURE)
+        {
+            int transformChosen = d->transformType->itemData(d->transformType->currentIndex()).toInt();
 
-    if (d->controller->checkLandmarks() == DTK_FAILURE)
-        return;
+            // Check that Affine process has enough landmarks
+            if (transformChosen == manualRegistration::AFFINE)
+            {
+                if ((d->controller->getPoints_Fixed()->count() < 4) || (d->controller->getPoints_Moving()->count() < 4))
+                {
+                    displayMessageError("Affine transformation needs 4 landmarks minimum by dataset");
+                    return;
+                }
+            }
 
-    this->setToolBoxOnWaitStatusForNonRunnableProcess();
+            this->setToolBoxOnWaitStatus();
 
-    d->process = new manualRegistration();
-    d->process->SetFixedLandmarks(d->controller->getPoints_Fixed());
-    d->process->SetMovingLandmarks(d->controller->getPoints_Moving());
-    d->process->setFixedInput(qobject_cast<medAbstractLayeredView*>(d->leftContainer->view())->layerData(0));
-    d->process->setMovingInput(qobject_cast<medAbstractLayeredView*>(d->rightContainer->view())->layerData(0));
-    d->process->setParameter(d->transformType->itemData(d->transformType->currentIndex()).toInt());
-    d->process->update(itkProcessRegistration::FLOAT);
+            d->process = dynamic_cast<medAbstractRegistrationProcess*> (dtkAbstractProcessFactory::instance()->create("manualRegistration"));
 
-    medRegistrationSelectorToolBox* toolbox = dynamic_cast<medRegistrationSelectorToolBox*>(selectorToolBox());
+            medRegistrationSelectorToolBox* toolbox = dynamic_cast<medRegistrationSelectorToolBox*>(selectorToolBox());
+            if(toolbox) // toolbox empty in Pipelines and not Registration workspace
+            {
+                toolbox->setProcess(d->process);
+            }
 
-    if(toolbox) // toolbox empty in Pipelines and not Registration workspace
-    {
-        toolbox->setProcess(d->process);
+            manualRegistration* process_Registration = dynamic_cast<manualRegistration*>(d->process);
+            process_Registration->SetFixedLandmarks(d->controller->getPoints_Fixed());
+            process_Registration->SetMovingLandmarks(d->controller->getPoints_Moving());
+            process_Registration->setFixedInput(qobject_cast<medAbstractLayeredView*>(d->leftContainer->view())->layerData(0));
+            process_Registration->setMovingInput(qobject_cast<medAbstractLayeredView*>(d->rightContainer->view())->layerData(0));
+            process_Registration->setParameter(transformChosen);
+
+            medRunnableProcess *runProcess = new medRunnableProcess;
+            runProcess->setProcess (d->process);
+            connect (runProcess, SIGNAL (success(QObject*)), this, SLOT(getOutputFromProcess()));
```

Comment:
You've changed ```d->process``` from a smart pointer to a raw one, meaning that this class is not participating anymore in the mutual ownership of this object, and yet it accesses this pointer in the ```getOutputFromProcess``` slot.  This means it relies on the process not being deleted by someone else before that, making many assumptions it has no control over. I think it should stay a smart pointer and simply be decremented in ```getOutputFromProcess``` once it is no longer needed.